cloudprovider/aws/aws_provider.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"encoding/json"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/rancher/agent/utilities/config"
+	"os"
+	"path"
+	"time"
+)
+
+const (
+	regionLabel           = "ec2.region"
+	availabilityZoneLabel = "ec2.availability_zone"
+	location              = "aws"
+	infoFile              = "info.json"
+	tempFile              = "temp.json"
+)
+
+type Info struct {
+	Label map[string]string `json:"label,omitempty"`
+}
+
+func CallProvider() {
+	infoPath := path.Join(config.StateDir(), location, infoFile)
+	tempPath := path.Join(config.StateDir(), location, tempFile)
+	select {
+	case <-time.After(time.Minute * 5):
+		return
+	default:
+		for {
+			if _, err := os.Stat(infoPath); err == nil {
+				return
+			}
+			time.Sleep(time.Second * 10)
+			session, err := session.NewSession()
+			if err != nil {
+				continue
+			}
+			client := ec2metadata.New(session)
+			document, err := client.GetInstanceIdentityDocument()
+			if err != nil {
+				continue
+			}
+			i := Info{}
+			i.Label = map[string]string{}
+			i.Label[regionLabel] = document.Region
+			i.Label[availabilityZoneLabel] = document.AvailabilityZone
+			bytes, err := json.Marshal(i)
+			if err != nil {
+				continue
+			}
+			os.MkdirAll(path.Join(config.StateDir(), location), 0755)
```

Comment:
Don't put "aws" in the path. It should not be `/var/lib/rancher/state/aws/config.json`. It should be `/var/lib/rancher/state/cloud_provider.json`.
This way the CloudProviderCollector can just generically read a single file without having to know about any of the providers